src/EPR.SubsidiaryBulkUpload.Application/Services/CompaniesHouseDownload/DownloadStatusStorage.cs

```diff
@@ -101,21 +100,39 @@ public async Task CreateCompaniesHouseFileDownloadLogAsync(string partitionKey)
                         DownloadStatus = null
                     });
                 }
-
-                var batchTransactions = new List<TableTransactionAction>();
-                batchTransactions.AddRange(entities.Select(e => new TableTransactionAction(TableTransactionActionType.Add, e)));
-                await tableClient.SubmitTransactionAsync(batchTransactions);
             }
-            catch (RequestFailedException ex)
+            else if (downloadsLog.Count != expectedFileCount)
             {
-                _logger.LogError(ex, "Cannot get or create table {TableName}", CompaniesHouseDownloadTableName);
+                for (int i = 1; i <= expectedFileCount; i++)
+                {
+                    var fileName = $"{PartialFilename}-{now.Year}-{now.Month:00}-01-part{i}_{expectedFileCount}.zip";
+                    var rowKey = RowKeyForThisMonth(now, i);
 
-                throw;
+                    if (!downloadsLog.Exists(x => x.RowKey == rowKey))
+                    {
+                        entities.Add(new CompaniesHouseFileSetDownloadStatus
+                        {
+                            PartitionKey = partitionKey,
+                            RowKey = RowKeyForThisMonth(now, i),
+                            DownloadFileName = fileName,
+                            DownloadStatus = null
+                        });
+                    }
+                }
             }
+
+            var batchTransactions = new List<TableTransactionAction>();
```

Comment:
This section only needs to run if (entities.Count > 0) - otherwise it will fail to run a transaction on an empty batch.